src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ReAllocCoTaskMemTests.cs

```diff
@@ -38,5 +38,59 @@ public void ReAllocCoTaskMem_Invoke_DataCopied()
                 Marshal.FreeCoTaskMem(p2);
             }
         }
+
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(100)]
+        [Theory]
+        public void ReAllocCoTaskMem_PositiveSize(int size)
+        {
+            IntPtr p = Marshal.ReAllocCoTaskMem(IntPtr.Zero, size);
+            Assert.NotEqual(IntPtr.Zero, p);
+
+            IntPtr p1 = Marshal.ReAllocCoTaskMem(p, size + 1);
+            Assert.NotEqual(IntPtr.Zero, p1);
+
+            IntPtr p2 = Marshal.ReAllocCoTaskMem(p1, 0);
+
+            // TODO: Behavior differs between Windows and Unix currently
```

Comment:
The aim of this PR to document the existing behavior and validate that the documentation is correct. I plan to unify the behavior in the other PR and remove these comments to highlight what have changed.